packages/opentelemetry-basic-tracer/src/Span.ts

```diff
@@ -0,0 +1,158 @@
+/**
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as types from '@opentelemetry/types';
+import {
+  randomSpanId,
+  randomTraceId,
+  INVALID_SPAN_CONTEXT,
+  isValid,
+} from '@opentelemetry/core';
+import { performance } from 'perf_hooks';
+import { TraceOptions } from '@opentelemetry/types';
+
+/**
+ * This class represents a span.
+ */
+export class Span implements types.Span {
+  private _spanContext: types.SpanContext = INVALID_SPAN_CONTEXT;
+  private _tracer: types.Tracer;
+  private _name: string;
+  private _parentId?: string;
+  private _kind: types.SpanKind;
+  private _attributes: types.Attributes = {};
+  private _links: types.Link[] = [];
+  private _events: types.Event[] = [];
+  private _status: types.Status = {
+    code: types.CanonicalCode.OK,
+  };
+  private _ended = false;
+  private _startTime: number;
+  private _duration = 0;
+
+  /** Constructs a new Span instance. */
+  constructor(
+    parentTracer: types.Tracer,
+    spanName: string,
+    options: types.SpanOptions
+  ) {
+    this._tracer = parentTracer;
+    this._name = spanName;
+    this._spanContext.spanId = randomSpanId();
```

Comment:
Could we provide the option of setting the `spanId` in the constructor? 

We have a use case for this in web tracing, where the server sends back the parent span ID for the server side render of the initial load, and then the client needs to create a span after the fact for the client side part of the initial load, that will be the parent of the server part.

Reply:
Does this also applies to `traceId`?

Reply:
@draffensperger I have filled new issue to discuss https://github.com/open-telemetry/opentelemetry-js/issues/138

Reply:
Regarding `traceId`, that is already implicitly passed in via `options`, so we are fine there!